src/sentry/api/endpoints/release_thresholds/utils/get_new_issue_counts.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List
+
+from django.db.models import CharField, Count, Q, Value
+
+from sentry.models.group import Group
+
+if TYPE_CHECKING:
+    from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+
+
+def get_new_issue_counts(
+    organization_id: int, thresholds: List[EnrichedThreshold]
+) -> dict[str, int]:
+    """
+    constructs a query for each threshold, filtering on project
+    """
+    queryset = None
+    for t in thresholds:
+        query = Q(
+            project__organization__id=organization_id,
+            project__id=t.project_id,
+        )
+        if t.environment:
+            # If we're filtering on environment, just use the first_release/first_seen from the groupenvironment so we don't need to cross reference tables
+            query &= Q(
+                groupenvironment__first_release__version=t.release,
+                groupenvironment__first_seen__range=(t.start, t.end),
+                groupenvironment__environment__name=t.environment.get("name"),
+            )
+        else:
+            query &= Q(
+                first_release__version=t.release,
+                first_seen__range=(t.start, t.end),
+                groupenvironment__isnull=True,
+            )
+        qs = (
+            Group.objects.filter(query)
+            .values("project__id", "first_release__version", "groupenvironment")
+            .annotate(
+                count=Count("*"),
+                threshold_id=Value(t.id, output_field=CharField()),
+            )
+        )
+        if queryset:
+            queryset.union(qs)
+        else:
+            queryset = qs
+
+    return {x.threshold_id: x.count for x in queryset}
```

Comment:
Should we just return the queryset for the caller, and allow the caller to determine when they want to execute and move data to memory? It could also help with future usage of the query as well! We could make it a private function for now and if it needs exporting allow the developer to change it at the time of their need

Reply:
Since this is just a utility method for the threshold health check, i'm not really inclined to care about returning the queryset.

If we ever find that we could use this elsewhere and we'd want to build on top of the query, then the future developer can modify this as needed, but for the sake of this feature, not necessary